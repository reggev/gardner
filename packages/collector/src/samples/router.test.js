const request = require('supertest');
const { app } = require('./../app');
const moment = require('moment');
const DataSource = require('./dataSource');
const { StatusCodes } = require('http-status-codes');
jest.mock('./dataSource.js');

const dummyBoards = [
  {
    id: 1,
    readings: [1, 2, 3],
  },
  {
    id: 2,
    readings: [4, 5, 6],
  },
];

describe('samples', () => {
  afterEach(() => {});
  test('post samples and get back the time until the next sample', async () => {
    const writeSamples = jest.spyOn(DataSource.prototype, 'writeSamples');
    const response = await request(app)
      .post('/samples')
      .send({ boards: dummyBoards });
    const [firstCall] = writeSamples.mock.calls;
    const [firstArg] = firstCall;
    expect(firstArg).toMatchSnapshot();
    expect(response.body).toMatchObject({ minutes: expect.any(Number) });
  });
  test('fail sending empty samples', async () => {
    const response = await request(app).post('/samples').send({ boards: [] });
    expect(response.status).toBe(StatusCodes.BAD_REQUEST);
  });
  test.each([
    { id: 1, readings: [] },
    { id: 1, reading: 1 },
    { id: 1, reading: ['1'] },
    { readings: [1] },
    { id: 1 },
  ])('fail sending bad samples $1', async board => {
    const response = await request(app)
      .post('/samples')
      .send({
        boards: [board],
      });
    expect(response.status).toBe(StatusCodes.BAD_REQUEST);
  });
});
