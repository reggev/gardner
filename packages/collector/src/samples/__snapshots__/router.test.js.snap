// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples post samples and get back the time until the next sample 1`] = `
Array [
  Object {
    "boardId": 1,
    "reading": 1,
    "sensorId": 0,
  },
  Object {
    "boardId": 1,
    "reading": 2,
    "sensorId": 1,
  },
  Object {
    "boardId": 1,
    "reading": 3,
    "sensorId": 2,
  },
  Object {
    "boardId": 2,
    "reading": 4,
    "sensorId": 0,
  },
  Object {
    "boardId": 2,
    "reading": 5,
    "sensorId": 1,
  },
  Object {
    "boardId": 2,
    "reading": 6,
    "sensorId": 2,
  },
]
`;
